app/components/ui/Section/OppEventCardSection.tsx

```diff
@@ -0,0 +1,110 @@
+import imageUrlBuilder from "@sanity/image-url";
+import React, { useEffect, useState } from "react";
+import { client } from "../../../sanity";
+import { OppEventCard } from "../Cards/OppEventCard";
+import { Button } from "../inline/Button/Button";
+import styles from "./OppEventCardSection.module.scss";
+
+export interface EventData {
+  slug: {
+    current: string;
+    _type: string;
+  };
+  image: {
+    asset: {
+      _ref: string;
+      _type: string;
+    };
+    _type: string;
+  };
+
+  /* Event Types*/
+  title?: string;
+  date?: {
+    endDate: Date;
+    startDate: Date;
+    _type: string;
+  };
+
+  /* Opportunity Types*/
+  name?: string;
+  start_date?: Date;
+  end_date?: Date;
+}
+
+export interface OppEventCardSectionProps {
+  sectionType: "opportunity" | "event";
+}
+/* 
+TODO: Future PRs
+  - Pull in Sanity data for header, description, button text
+  - Update Button to use global button component
+    - Update header button and view more button
+    - include link to opportunity/event page
+*/
+
+export const OppEventCardSection = (props: OppEventCardSectionProps) => {
+  const { sectionType } = props;
+  const [eventData, setEventData] = useState<EventData[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const builder = imageUrlBuilder(client);
+
+  useEffect(() => {
+    const fetchEventData = async () => {
+      const fields = {
+        event: "title, slug, date, image",
+        opportunity: "name, slug, start_date, end_date, image",
+      };
+      const query = `*[_type == "${sectionType}"] | order(_createdAt desc)[0...4] { ${fields[sectionType]} }`;
+
+      const result: EventData[] = await client.fetch(query);
+      setEventData(result);
+    };
+
+    setIsLoading(true);
+    fetchEventData();
+    setIsLoading(false);
```

Comment:
I think setIsLoading will be set to false before the event data finishes fetching. I believe if you call setIsLoading(false) inside the fetching function after the fetch, it'll render the states in the proper order!